### PR TITLE
pallet-democracy  multi instance [A*]

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
Signed-off-by: Dzmitry Lahoda <dzmitry@lahoda.pro>

This PR makes democracy to be multi instance. Each can have its own assets lock and root to enact/schedule.

Changed are storage and behaviour compatible, but not runtime API configuration (can configure same behavior, but will not compile old configuraiton).

polkadot companion:https://github.com/paritytech/polkadot/pull/5783

✄ -----------------------------------------------------------------------------

Thank you for your Pull Request! 🙏

Before you submit, please check that:

- [x] **Description:** You added a brief description of the PR, e.g.:
  - What does it do?
  - What important points should reviewers know?
  - Is there something left for follow-up PRs?
- [ ] **Labels:** You labeled the PR appropriately if you have permissions to do so:
  - [X] `A*` for PR status (**one required**)
  - [ ] `B*` for changelog (**one required**)
  - [ ] `C*` for release notes (**exactly one required**)
  - [ ] `D*` for various implications/requirements
  - [ ] Github project assignment
- [X] **Related Issues:** You mentioned a related issue if this PR is related to it, e.g. `Fixes #228` or `Related #1337`.
- [ ] **2 Reviewers:** You asked at least two reviewers to review. If you aren't sure, start with GH suggestions.
- [X] **Style Guide:** Your PR adheres to [the style guide](https://github.com/paritytech/substrate/blob/master/docs/STYLE_GUIDE.md)
  - In particular, mind the maximal line length of 100 (120 in exceptional circumstances).
  - There is no commented code checked in unless necessary.
  - Any panickers in the runtime have a proof or were removed.
- [X] **Runtime Version:** You bumped the runtime version if there are breaking changes in the **runtime**.
- [X] **Docs:** You updated any rustdocs which may need to change.
- [X] **Polkadot Companion:** Has the PR altered the external API or interfaces used by Polkadot?
  - [x] If so, do you have the corresponding Polkadot PR ready?
  - [ ] Optionally: Do you have a corresponding Cumulus PR?

Refer to [the contributing guide](https://github.com/paritytech/substrate/blob/master/docs/CONTRIBUTING.adoc) for details.

After you've read this notice feel free to remove it.
Thank you!

✄ -----------------------------------------------------------------------------

### CODE REVIEW

**Comment by dzmitry-lahoda** [comment]
how one may ask to run ci again after companion fixes?

**Comment by dzmitry-lahoda** [comment]
master make step forward, need to update companion

File: frame/democracy/src/lib.rs [hunk 1]
```diff
@@ -369,74 +376,96 @@ pub mod pallet {
...
+
+		/// Runtime unique identifier for locking currency.
+		/// May be equivalent to PalletId.
+		#[pallet::constant]
+		type DemocracyId: Get<LockIdentifier>;
+
+		/// An origin on behalf of which the preimage calls are dispatched.
+		/// You may consider set it to be same as `Self::EnsureRoot`.
+		/// May be runtime wide or per set of pallets.
+		type ProtocolRoot: Get<<Self as Config<I>>::Origin>;
```
Comment (kianenigma) [inline frame/democracy/src/lib.rs:393]: Why are these types added, if the purpose of the PR is to only make the pallet instanciable?

Comment (dzmitry-lahoda) [reply frame/democracy/src/lib.rs:393]: Purpose of pallet to make it meaningfully instantiable. By meaning I mean that second democracy should be able to execute on behalf of non system::Orign::Root (if 2 democracies execute as same origin, then why there are 2)? So you can execute via this on behalf of some account, for example anonymous proxy account without delegation or account derived from pallet id. So can do relay(parachain(protocol)) each has its own ""root"" and democracy. 

Same for currency, if we vote with same currency, why there is another democracy? So protocol can his own root and own  currency and own democracy.

**Comment by dzmitry-lahoda** [comment]
Need to bring it up to date. So not sure when democracy 2.0 will be released and may be patch it.

**Pull request closed without merging**